host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -204,6 +205,22 @@ public HostControllerRegistrationHandler(ManagementChannelHandler handler, Domai
          */
         ModelNode executeReadOnly(ModelNode operation, Resource model, OperationStepHandler handler, ModelController.OperationTransactionControl control);
 
+        /**
+         * Registers a slave host controller. A non-exclusive read lock is held for the duration of this registration
+         * process, blocking writes, but allowing the initiation of additional read locks.
+         * @param operationId - the operation ID for this registration
+         * @param hostInfo - the registering host's HostInfo
+         * @throws OperationFailedException - if the lock acquisition fails
+         */
+        void beginRegistration(final Integer operationId, final HostInfo hostInfo) throws OperationFailedException;
```

Comment:
Yeah, good point. I had more code in here at one point, but doing so involves exposing a lot of stuff owned by HCRH at the moment, like RegistrationContext etc.

I'm thinking of just calling them acquireReadlock() / releaseReadlock() now.